### PR TITLE
fix(design): Update critical color for dark mode

### PR DESCRIPTION
**Pull request by chris-at-jobber**
## Motivations

For the same reasons that `destructive` was made lighter and less saturated for our iterations on dark mode, critical should follow suit.

### Changed

- critical's use of red to match destructive in dark mode

### Fixed

- critical iconography and text is more readable in dark mode
- instances where critical text was being used for instances that should be destructive now use destructive

## Testing

Some good examples would be testing form validation

Changes can be
[tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components-native/src/Typography/Typography.tsx [hunk 1]
```diff
@@ -334,12 +335,14 @@ export type TextColor =
 export type TextVariation =
   | "success"
   | "interactive"
+  | "interactiveSubtle"
```
Comment (chris-at-jobber) [inline packages/components-native/src/Typography/Typography.tsx:338]: Extending the semantic variations here is part of https://jobber.atlassian.net/browse/JOB-97200

File: packages/design/src/tokens/dark.tokens.json [hunk 2]
```diff
@@ -51,7 +51,7 @@
             }
         },
         "critical": {
-            "$value": "{color.base.red-.600}"
+            "$value": "{color.base.red-.500}"
```
Comment (chris-at-jobber) [inline packages/design/src/tokens/dark.tokens.json:54]: This is the key change here - destructive and critical now match

image

File: packages/components/src/DataList/components/DataListAction/DataListAction.tsx [hunk 3]
```diff
@@ -23,7 +23,7 @@ export function DataListAction<T extends DataListObject>({
     return null;
   }
 
-  const color = destructive ? "critical" : "heading";
+  const color = destructive ? "destructive" : "heading";
```
Comment (chris-at-jobber) [inline packages/components/src/DataList/components/DataListAction/DataListAction.tsx:26]: image

Comment (chris-at-jobber) [reply packages/components/src/DataList/components/DataListAction/DataListAction.tsx:26]: Local URL for [DataList](http://localhost:6005/?path=/story/components-lists-and-tables-datalist-web--basic)

**Review by Aiden-Brine — APPROVED** [review]

**Pull request merged**